apps/explorer/src/app/components/proposals/proposals-table.tsx

```diff
@@ -0,0 +1,221 @@
+  NetworkParams,
+  t,
+  useNetworkParams,
+} from '@vegaprotocol/react-helpers';
+import { ProposalStateMapping } from '@vegaprotocol/types';
+import BigNumber from 'bignumber.js';
+import { DApp, TOKEN_PROPOSAL, useLinks } from '@vegaprotocol/environment';
+import { BREAKPOINT_MD } from '../../config/breakpoints';
+import { JsonViewerDialog } from '../dialogs/json-viewer-dialog';
+
+type ProposalTermsDialog = {
+  open: boolean;
+  title: string;
+  content: unknown;
+};
+type ProposalsTableProps = {
+  data: ProposalListFieldsFragment[] | null;
+};
+export const ProposalsTable = ({ data }: ProposalsTableProps) => {
+  const { params } = useNetworkParams([
+    NetworkParams.governance_proposal_market_requiredMajority,
+  ]);
+  const tokenLink = useLinks(DApp.Token);
+  const requiredMajorityPercentage = useMemo(() => {
+    const requiredMajority =
+      params?.governance_proposal_market_requiredMajority ?? 1;
+    return new BigNumber(requiredMajority).times(100);
+  }, [params?.governance_proposal_market_requiredMajority]);
+
+  const gridRef = useRef<AgGridReact>(null);
+  useLayoutEffect(() => {
+    const showColumnsOnDesktop = () => {
+      gridRef.current?.columnApi.setColumnsVisible(
+        ['voting', 'cDate', 'eDate'],
+        window.innerWidth > BREAKPOINT_MD
+      );
+      gridRef.current?.columnApi.setColumnWidth(
+        'actions',
+        window.innerWidth > BREAKPOINT_MD ? 221 : 155
+      );
+    };
+    window.addEventListener('resize', showColumnsOnDesktop);
+    return () => {
+      window.removeEventListener('resize', showColumnsOnDesktop);
+    };
+  }, []);
+
+  const [dialog, setDialog] = useState<ProposalTermsDialog>({
+    open: false,
+    title: '',
+    content: null,
+  });
+
+  return (
+    <>
+      <AgGrid
+        ref={gridRef}
+        rowData={data}
+        getRowId={({ data }: { data: ProposalListFieldsFragment }) =>
+          data.id || data.rationale.title
+        }
+        overlayNoRowsTemplate={t('This chain has no markets')}
+        domLayout="autoHeight"
+        defaultColDef={{
+          flex: 1,
+          resizable: true,
+          sortable: true,
+          filter: true,
+          filterParams: { buttons: ['reset'] },
+          autoHeight: true,
+        }}
+        suppressCellFocus={true}
+        onRowClicked={({ data, event }: RowClickedEvent) => {
+          if (
+            (event?.target as HTMLElement).tagName.toUpperCase() !== 'BUTTON'
+          ) {
+            const proposalPage = tokenLink(
+              TOKEN_PROPOSAL.replace(':id', data.id)
+            );
+            window.open(proposalPage, '_blank');
+          }
+        }}
+      >
+        <AgGridColumn
+          colId="title"
+          headerName={t('Title')}
+          field="rationale.title"
+        />
+        <AgGridColumn
+          colId="type"
+          maxWidth={180}
+          headerName={t('Type')}
+          field="terms.change.__typename"
+        />
+        <AgGridColumn
```

Comment:
**question (non-blocking)**:  Is it possible to prefill the filter selection on this? I think it's currently treating it like a string, where really it's an enumb of (something like) `[Enacted, Open]`. On the network I'm looking at, all proposals are `Enacted` and that gives me no clue what the alternatives would be if I wanted to filter to just the Active ones.

<img width="300" alt="Screenshot 2023-02-17 at 09 58 16" src="https://user-images.githubusercontent.com/6678/219613415-42f744b7-afbe-4a9c-8075-0eff2e03d85d.png">

Reply:
There's an out of the shelf filter like that in Enterprise edition of Ag Grid. For our current Community edition we'd need to make our own.